Tests/Common/SimulatedDevice.cs

```diff
@@ -67,29 +67,25 @@ public SimulatedDevice(TestDeviceInfo testDeviceInfo, uint frmCntDown = 0, uint
             FrmCntUp = frmCntUp;
         }
 
-        public LoRaPayloadJoinRequest CreateJoinRequest()
+        public LoRaPayloadJoinRequest CreateJoinRequest(string appkey = null)
         {
             var devNonceBytes = new byte[2];
             using var random = RandomNumberGenerator.Create();
             random.GetBytes(devNonceBytes);
             DevNonce = DevNonce.Read(devNonceBytes);
 
             TestLogger.Log($"[{LoRaDevice.DeviceID}] Join request sent DevNonce: {DevNonce:N} / {DevNonce}");
-#pragma warning disable CA1508 // Avoid dead conditional code
             var joinRequest = new LoRaPayloadJoinRequest(LoRaDevice.AppEUI, LoRaDevice.DeviceID, DevNonce);
-#pragma warning restore CA1508 // Avoid dead conditional code
-            joinRequest.SetMic(LoRaDevice.AppKey);
+            joinRequest.SetMic(appkey is null ? LoRaDevice.AppKey : appkey);
```

Comment:
nit: can be simplified to:

```suggestion
            joinRequest.SetMic(appkey ?? LoRaDevice.AppKey);
```